### DISCUSSION TITLE
Storage: createSignedUrl for upload

### DISCUSSION THREAD

**Discussion by ilyabo** [supabase/supabase#10289] [Feature Requests]
I need to use my own upload UI on the client which shows upload progress. Is there a way to create a signed URL for upload? I haven't found how to do it with `supabase-js` so far.  Calling `createSignedUrl` with an object which doesn't exist results in an error. 

It's possible using S3 libraries via `PutObjectCommand`:
```
const url = await getSignedUrl(
  S3_URL,
  new (upload ? PutObjectCommand : GetObjectCommand)({
    Bucket: S3_BUCKET,
    Key: `${fname}`,
  }),
  {expiresIn: 15 * 60},
)
```
Google Storage API also has an option for that called  `action`:

```
  const options: GetSignedUrlConfig = {
    version: 'v4',
    action: upload ? 'write' : 'read',
    expires: Date.now() + 15 * 60 * 1000,
  };

  const url = await storage
      .bucket(GOOGLE_STORAGE_BUCKET_NAME!)
      .file(`${fname}`)
      .getSignedUrl(options)
  )[0]
      
```

**Comment by devidevio** [comment]
Check this out, it should fit your case:
* https://github.com/supabase/storage-api/issues/23#issuecomment-973277262

**Comment by inian** [comment]
Hi, this has been implemented https://supabase.com/docs/reference/javascript/storage-from-uploadtosignedurl
